test/common/mongo/proxy_test.cc

```diff
@@ -236,6 +237,20 @@ TEST_F(MongoProxyFilterTest, MultiGet) {
   EXPECT_EQ(1U, store_.counter("test.collection.test.query.multi_get").value());
 }
 
+TEST_F(MongoProxyFilterTest, MaxTime) {
+  EXPECT_CALL(*filter_->decoder_, onData(_))
+      .WillOnce(Invoke([&](Buffer::Instance&) -> void {
+        QueryMessagePtr message(new QueryMessageImpl(0, 0));
+        message->fullCollectionName("db.test");
+        message->flags(0b1110010);
+        message->query(Bson::DocumentImpl::create()->addInt32("$maxTimeMS", 100));
+        filter_->callbacks_->decodeQuery(std::move(message));
+      }));
+  filter_->onData(fake_data_);
+
+  EXPECT_EQ(0U, store_.counter("test.op_query_no_max_time").value());
```

Comment:
Can you add the inverse of this stat check to some other test case so that you actually cover the stat increment.